x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/hooks/use_index_data.ts

```diff
@@ -93,11 +96,32 @@ export const useIndexData = (indexPattern: IndexPattern, query: any): UseIndexDa
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [indexPattern.title, JSON.stringify([query, pagination, sortingColumns])]);
 
+  const fetchColumnChartsData = async function () {
+    const columnChartsData = await fetchChartsData(
+      indexPattern.title,
+      ml,
+      query,
+      columns.filter((cT) => dataGrid.visibleColumns.includes(cT.id))
+    );
+    dataGrid.setColumnCharts(columnChartsData);
+  };
```

Comment:
you probably should catch an error here and show a toast notification in case the fetch has failed?

Reply:
Fixed in f9a1399.